src/download_logfile.py

```diff
@@ -0,0 +1,227 @@
+                                else:
+                                    entry.size = ("{:.2f}".format(size/1024)).rjust(8) + " kB"
+                                entry.datafile = path
+                            self.UpdateEntry(entry, enc)
+        index = 0
+        if self.args.latest:
+            print("latest:")
+            for idx, entry in enumerate(self.log_list):
+                if entry.date > self.log_list[index].date:
+                    index = idx
+                elif entry.date == self.log_list[index].date:
+                    if entry.time > self.log_list[index].time:
+                        index = idx
+            print(self.log_list[index].date + "/" + self.log_list[index].time)
+        else:
+            # Interactive selection
+            print("logs:")
+            for idx, entry in enumerate(self.log_list):
+                output = str(idx).rjust(5) + ": " + (entry.date+"/"+entry.time).ljust(25) + " " + str(entry.size)
+                if entry.enc:
+                    output = output + " (encrypted)"
+                    if entry.keyfile is None:
+                        output = output + " KEYFILE MISSING"
+                print(output)
+            index = input("\nSelect log file to download: ")
+        return index
+
+    async def download_file(self, file, date, dir):
+        previous_progress = 0
+        async for progress in self.mav.ftp.download(file, dir):
+            new_progress = round((progress.bytes_transferred/progress.total_bytes)*100)
+            if new_progress != previous_progress:
+                sys.stdout.write(f"\r{new_progress} %")
+                sys.stdout.flush()
+            previous_progress = new_progress
+        print()
+        filename = [f for f in os.listdir(dir) if os.path.isfile(os.path.join(dir, f))]
+        infilebase, infileext = filename[0].split('.')
+        outfile = "log-" + date + "T" + infilebase.replace('_', '-') + "Z." + infileext
+        print("file: " + outfile)
+        os.rename(os.path.join(dir, filename[0]), os.path.join(self.args.dir, outfile))
+
+
+    async def download_logfile(self, index):
+        with tempfile.TemporaryDirectory(prefix="logfile_dl_") as tmpdir:
+            entry = self.log_list[index]
+            await self.download_file(entry.datafile, entry.date, tmpdir)
+            if entry.keyfile is not None:
+                await self.download_file(entry.keyfile, entry.date, tmpdir)
+
+    async def run(self):
+        workdir = os.getcwd()
+        parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter,description=__doc__)
+        parser.add_argument('-a', '--address', action="store", help='Address to connect. (e.g serial:///dev/ttyACM0, udp:192.168.200.101:14540, tcp://:5760)', default='tcp://:5760')
+        parser.add_argument('-d', '--dir', action="store", help='Output directory (default .)', default=workdir)
```

Comment:
I have a small concern regarding invoking this from hypervisor side with `$ fog tool download_logfile.py`.

It runs the container with `$ docker run --rm -it`: https://github.com/tiiuae/fog_hyper/blob/7bfdfb5ae78adf386afb24080772815b8c2d12b0/pkg/foghyper/tools.go#L41

So when this script downloads something to any dir (probably the default workdir, i.e. `/fog-tools`), and the container exits (when this script finishes), all of the FS changes inside the container are lost.

It might be a common task to transfer some data out of this container (also interactively, think running `$ fog tool bash` to get a shell inside this container), so should we bind mount something like `/var/lib/foghyper/tools-data` to `/tools-data` inside the container? I.e. `$ docker run -v /var/lib/foghyper/tools-data:/tools-data ...` and then one could use `$ fog tool download_logfile.py --dir=/tools-data` so the log file is saved and also accessible from host?

Additional question, should dir `--dir=/tools-data` be made the default? Or should we make container workdir default to  `/tools-data`, so any files written to workdir are automatically persisted?

Reply:
For context: `$ fog debug traffic ...` does something similar: it bind mounts a dir inside the container so packet capture files can be easily persisted back to the host system: https://github.com/tiiuae/fog_hyper/blob/7bfdfb5ae78adf386afb24080772815b8c2d12b0/pkg/foghyper/debugtraffic.go#L95

Reply:
For generic usage, it would be useful to have default output directory as "." so that it would work predictable also when calling manually. I vote for having the workdir set as /tools-data

Reply:
On the other hand, if the workdir is not pointing to the script dir, the list of tools is not visible so easily. Either there should be some help/tools command to show available scripts (and some initial print to tell to use that help command), or then workdir should be point to script directory and the default output would be the tools-data. Not sure anymore, which one would be the best.

Reply:
> if the workdir is not pointing to the script dir, the list of tools is not visible so easily. Either there should be some help/tools command to show available scripts

I think this feature is already supposed to address that use case:

(https://github.com/tiiuae/fogsw_tools/blob/cea0076d46552e5969ab8490d1fddecfc5671a29/Dockerfile#L26)

That means that `$ fog tool` will run `$ ls -1 /fog-tools` inside the container.

Do you think this is sufficient? `$ fog tool --help` will not list those scripts currently (but can be made to)

Reply:
Ok, I think it is ok that way.
So, if workdir would just point to /tools-data, then the downloader script could just use dir '.' by default.

Reply:
Yes, I think that could be an optimal solution